### DISCUSSION TITLE
Why no `this.`?

### DISCUSSION THREAD

**Discussion by Gammer0909** [dotnet/roslyn#69440] [Q&A]
In the style guide for contributing, it mentions:
> We avoid `this.` unless absolutely necessary.

Why is this? (Haha) Is it a performance-killer? If anyone could help, that'd be great.

**Comment by CyrusNajmabadi** [comment]
No.  It's purely stylistic.  There is no runtime difference between the two.  We just state this as the preference so code can be consistent.  Note: we are very inconsistent with this rule, so even we don't take it very seriously :)

I will personally state that i use `this.` when i find it adds to readability.  I never use it to preface an `_` prefaced member.  So i never to `this._count++`.  it would always be `_count++`.  But i definitely use `this.` if i feel it aids when calling properties or members.

**Reply by Gammer0909** [reply]
Huh, interesting. Thanks again, sorry for the dumb questions, I'm still learning.

**Reply by CyrusNajmabadi** [reply]
Not a dumb question at all :)

Always feel free to ask.  We also have good discords where questions and participation is welcome.  Like:

https://discord.com/channels/143867839282020352/598678594750775301 and
https://discord.com/channels/732297728826277939/732297994699014164

Thanks!

**Reply by Gammer0909** [reply]
Thanks a lot for all the resources 😃
